### DISCUSSION TITLE
CreateMap and CreateProjection for same source / target types

### DISCUSSION THREAD

**Discussion by nstdspace** [LuckyPennySoftware/AutoMapper#4056] [NA]
As the upgrade guide [clearly states](https://docs.automapper.org/en/latest/12.0-Upgrade-Guide.html#equivalent-settings-overwrite-each-other) and several stack overflow questions point out, it is not possible to define `CreateMap` and `CreateProjection` for the same pair of source- and target-types.

Why is that?

We have often the case, that a mapping is done sometimes "inbetween some service logic", where a direct instance to be mapped is given (i.e., only Map<> can be used), and sometimes directly succeeding a database query (i.e., ProjectTo<> would provide a significat performance boost but we are using Map<> instead).
Is this not a common requirement?

**Comment by lbargaoanu** [comment]
It is. And that's what `CreateMap` does :) So if you need `Map`, you have no use for `CreateProjection`.

**Reply by jbogard** [reply]
And to clarify *why*, `CreateMap` builds the `Map` expression tree at config-time. The projection expression tree is created with `ProjectTo` then cached. If you never call `Map` then that extra expression tree, which is way more complicated, wastes resources.

`CreateProjection` was added for just that case.
